modules/local/simplifyiridajson/main.nf

```diff
@@ -2,7 +2,14 @@ process SIMPLIFY_IRIDA_JSON {
     tag "$meta.id"
     label 'process_single'
 
-    container 'docker.io/python:3.9.17'
+    // Container directive is intentionally using the "override_configure_container_registry" as an example:
+    // How to keep a non-biocontainer/quay.io default, see nextflow.config for
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    'https://depot.galaxyproject.org/singularity/python:3.10' :
+    task.ext.override_configured_container_registry != false ?
+    'docker.io/python:3.10' :
+    'biocontainers/python:3.10' }"
```

Comment:
In this particular case example, I think you should remove `biocontainers` from the container string and also the expression for the container on `depot.galaxyproject.org`. That is something like:

```groovy
    container "${ task.ext.override_configured_container_registry != false ?
    'docker.io/python:3.10' :
    'python:3.10' }"
```

My reasoning being that `biocontainers/python:3.10`, when combined with the default `quay.io` becomes `quay.io/biocontainers/python:3.10` (i.e., Python from quay.io). But, returning `docker.io/python:3.10` will always return the dockerhub (docker.io) container for Python. These both come from different container registries.

For removing the `depot.galaxyproject.org`, this is because galaxyproject.org is only storing the `quay.io/biocontainers` version of the singularity Python container, not the `docker.io` one.

The intention of this ternary operator is to be able to handle situations where you want to default to downloading a container from `docker.io` (i.e., `docker.io/python:3.10`) for this process. But, you want to allow users/admins to override the overall container registry to point to a private container registry (e.g., for running in Azure). Since the default container registry is `quay.io` for this pipeline, we need to explicitly return `docker.io/python.3.10` when the pipeline is run under default conditions, but if a private container registry is configured, then `task.ext.override_configured_container_registry` will be set to `false` and so we return `python:3.10`, which then uses the private container registry `private-registry.io/python:3.10`.

I think the change above better reflects an example of the intended scenario where this will be used. Let me know if you have other questions.

Reply:
My only issue with removing `biocontainer` is that if someone does want use the `docker.registry` for this process by making `process.ext.override_configured_container_registry = false` `quay.io/python:3.10` won't work.  Should we create a `private-registry.io` param or process, and set the default to `quay.io/biocontainer` OR
Does it make sense to change `docker.registry = 'quay.io/biocontainer'`. Does that make sense?

Reply:
So, `docker.registry` is always set (by default to `quay.io`).

The only time someone would change the default for `docker.registry` would be to set a private container registry (e.g., `docker.registry = private-registry.io`). In this case, they would be expected to also set `process.ext.override_configured_container_registry = false`, at which point the container used here would be `private-registry.io/python:3.10`.

There is no scenario I can really think of where someone would want to set `process.ext.override_configured_container_registry = false` without also overriding `docker.registry`.

Reply:
Initially, I was thinking of a scenario where someone didn't want to use `docker.io` but rather `quay.io` but the way you described it, I now agree it is unlikely. They'd have to be real `quay.io` stans 😜. I will make the changes! Thanks for taking the time explain.

Reply:
fa55e4d

Reply:
Thanks so much Steven 😄